rfc/system/0000-flyteadmin-rbac.md

```diff
@@ -0,0 +1,203 @@
+The authorization config will be used to configure the authorization behavior. The first thing it describe is a way to
+resolve roles from the user's identity context. Ideally this should be flexible enough to resolve a role from different
+elements of a token to support various use cases (ie. standard and custom JWT claims). 
+
+```yaml
+authorization:
+  roleResolutionStrategies: # one or more can be configured
+    - scopes # attempts to use token scopes as roles
+    - claims # attempts to use token claim values as roles
+    - userID # attempts to use the user ID as the role
+  
+  # below is an example of what configuring custom claims might look like  
+  claimRoleResolver: # claim based role resolution requires additional configuration
+    - key: groups # this is the key to look for in the token claims
+      type: list # this declares the structure of the value. parse value as comma separated string
+    - key: group # this is the key to look for in the token claims
+      type: string # this declares the structure of the value. parse value as single string
+```
+
+The second part of the authorization will include exceptions. There are some RPCs which should not be authorized. 
+```yaml
+authorization:
+  methodBypassPatterns: # ideally these should be enabled by default
+    - "/grpc.health.v1.Health/.*" # health checking for k8s
+    - "/flyteidl.service.AuthMetadataService/.*" # auth metadata used by other Flyte services
+```
+
+The last part of the authorization will declare the authorization policies. The authorization policies will include the 
+name of the role and will include a list of permitting rules. Each rule must specify an RPC pattern which is
+regex statement that matches the target gRPC method. Each rules includes optional project and domain isolation/filtering 
+fields. If the project and domain fields are not included it is implied that the rule applies to all projects and domains.
+Below is an example authorization configuration with a variety of use cases of both RBAC and project and domain isolation/filtering.
+```yaml
+authorization:
+  policies: # policies is a list instead of a map since viper map keys are case insensitive
+    - name: admin
+      rules: # list of rules
+        - name: "allow all" # name / description of the rule
+          methodPattern: ".*" # regex pattern to match on gRPC method
+    - name: read-only
+      rules:
+        - name: "read everything"
+          methodPattern: "Get.*|List.*"
+    - name: mapping-team
+      rules:
+        - name: "r/w for the mapping project in dev only"
+          methodPattern: ".*"
+          project: mapping # access to only the mapping project
+          domain: development # access to only the development domain within the mapping project
+    - name: ci
+      rules:
+        - name: "r/w for every project in production"
+          methodPattern: ".*"
+          domain: production # you can wildcard project and declare domain level access
```

Comment:
does the same work for wild carding domain if you want, for example, mapping team write access?

Reply:
Yes absolutely